### PR TITLE
#2315 Add peer deps

### LINKED ISSUE #2315: Getting rid of warnings while installing packages
At current head commit `06003a5ced1ac8ce415f04643aba672de05cd037` we have these warnings

Would be nice if we can mitigate a few of them. 
Warnings are usually there for reasons to raise awareness something might be bug risk in the future.
image

### PR DESCRIPTION
**Pull request by preschian**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [ ] Bugfix
- [ ] Feature
- [x] Refactoring

### What's new?

- [x] PR closes #2315
- [x] Getting rids of some warnings by installing packages in devDependencies

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

### Optional

- [ ] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [ ] I've tested PR on mobile and everything seems works
- [ ] I found edge cases
- [ ] I've written some unit tests 🧪

### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=DY4SQF2iD456tH89aQtz5wv1EV3BbSW8wKKuMcwbmXaj1pM)

### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

| Before  | After |
| ------------- | ------------- |
| Screen Shot 2022-03-12 at 10 13 56 AM | Screen Shot 2022-03-12 at 10 14 31 AM |

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -114,6 +114,12 @@
     "@nuxt/typescript-build": "^2.1.0",
     "@nuxtjs/pwa": "^3.3.5",
     "@open-wc/webpack-import-meta-loader": "^0.4.7",
+    "@polkadot/api": "7.10.1",
+    "@polkadot/keyring": "^8.4.1",
+    "@polkadot/networks": "^8.4.1",
+    "@polkadot/ui-settings": "0.89.3",
+    "@polkadot/util": "^8.4.1",
+    "@polkadot/util-crypto": "^8.4.1",
```
Comment (vikiival) [inline package.json:122]: Not fan of this tbh.

Isn't it better to update the [@kodadot1 packages](https://github.com/kodadot/packages)

**Review by vikiival — COMMENTED** [review]
🤷

File: package.json (hunk 1 above)
Comment (preschian) [reply package.json:122]: sure, we can update `@kodadot1/minimark`. afaik, the rest of the packages still got the warning messages:
- vue-apollo -> graphql-tag
- @polkadot/extension-dapp. source: https://github.com/polkadot-js/extension/blob/master/packages/extension-dapp/package.json#L28-L32
  - @polkadot/api
  - @polkadot/util
  - @polkadot/util-crypto

etc...

there is an explicit message if the repo using `pnpm` 
Screen Shot 2022-03-11 at 10 58 27 PM

**Review by vikiival — CHANGES_REQUESTED** [review]
Let's update minimark to avoid inconsitsencies

**Comment by vikiival** [comment]
Updated `minimark@0.0.1-rc.6`

**Review by vikiival — CHANGES_REQUESTED** [review]
Resolve the conversation first

File: package.json (hunk 1 above)
Comment (preschian) [reply package.json:123]: for now, the app may be still running perfectly. but not good in the long run if we keep the warning. turns out:
- `apollo-boost` is deprecated https://www.apollographql.com/docs/react/migrating/apollo-client-3-migration/#apollo-boost
- `@nuxtjs/apollo` is unmaintained and there is a security concern if I look at the details in snyk https://snyk.io/advisor/npm-package/@nuxtjs/apollo#security

those packages technically tested only in graphql <= 15, for now maybe still run with graphql 16. in the future, not sure still able to run with newer graphql or not

**Comment by yangwao** [comment]
pinging on this, seems minimark rc6 was published, updating it to this PR 👀

**Comment by preschian** [comment]
updated, ci/cd looking good (all green)
cc @yangwao @vikiival

**Comment by preschian** [comment]
sorry for make noisy in this PR 🙏🏻

`node-ipc` issue yesterday is a false alarm. `@vue/cli` is not supposed to be used in the `nuxt` project, so I removed them also in this PR

afaik, `@vue/cli` is used in `vue.config.js` not `nuxt.config.js`

**Comment by preschian** [comment]
revert `@polkadot/*` from `package.json`. changes in this PR basically:
- remove some peer dependencies warning
- remove some unused packages
- added `vue-shim.d.ts` based on this https://typescript.nuxtjs.org/guide/setup

final results:
Screen Shot 2022-03-23 at 12 30 44 PM

**Comment by preschian** [comment]
revert `@polkadot/*` from `package.json`. changes in this PR basically:
- remove some peer dependencies warning
- remove some unused packages
- added `vue-shim.d.ts` based on this https://typescript.nuxtjs.org/guide/setup

final results:
Screen Shot 2022-03-23 at 12 30 44 PM

**Comment by yangwao** [comment]
pay 100 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $100 @ 165.41 USD/KSM ~ 0.605 $KSM
🧗 [DY4SQF2iD456tH89aQtz5wv1EV3BbSW8wKKuMcwbmXaj1pM](https://kusama.subscan.io/account/DY4SQF2iD456tH89aQtz5wv1EV3BbSW8wKKuMcwbmXaj1pM)
🔗 [0x7b0f1242d7b5696eb88ebb985ce271adbc5ae4bc8da36180a1e2abb6c4800216](https://kusama.subscan.io/extrinsic/0x7b0f1242d7b5696eb88ebb985ce271adbc5ae4bc8da36180a1e2abb6c4800216)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Comment by yangwao** [comment]
let's merge it :)

**Pull request merged**